src/zk.py

```diff
@@ -220,14 +220,22 @@ def _acquire_lock(self, name, allow_queue, timeout, read_lock=False):
                 logging.warning('%s lock is already taken by %s.', name[0].upper() + name[1:], contenders[0])
                 return False
         try:
-            return lock.acquire(blocking=True, timeout=timeout)
+            acquired = lock.acquire(blocking=True, timeout=timeout)
+            if not acquired:
+                logging.warning('Unable to acquire lock "%s", but not because of timeout...', name)
         except LockTimeout:
             logging.warning('Unable to obtain lock %s within timeout (%s s)', name, timeout)
-            return False
+            acquired = False
         except Exception:
             for line in traceback.format_exc().split('\n'):
                 logging.error(line.rstrip())
-            return False
+            acquired = False
+        if not acquired and release_on_fail:
+            logging.debug('Try to release and delete lock "%s", to recreate on next iter', name)
+            self.release_lock(name)
```

Comment:
Actually, `release_lock()` can throw exceptions in case when there is some connection problem to ZK, and we are not catching it here.
IDK where it would be caught, but it seems like a wrong behaviour.